src/DssProxyActionsCharter.sol

```diff
@@ -372,12 +377,11 @@ contract DssProxyActionsCharter is Common {
         bytes32 ilk,
         address daiJoin
     ) external {
-        address vat = CharterLike(charter).vat();
         address urp = CharterLike(charter).getOrCreateProxy(address(this));
         (, uint256 art) = VatLike(vat).urns(ilk, urp);
 
         // Joins DAI amount into the vat
-        daiJoin_join(daiJoin, _getWipeAllWad(vat, urp, ilk));
+        daiJoin_join(daiJoin, _getWipeAllWad(urp, ilk));
```

Comment:
Do we maybe want to optimize for not doing a join if `_getWipeAllWad` returns zero? I guess it is an amortization question that does not have an obvious answer, so fine either way.

Reply:
I think the case where it returns zero is by far less likely, so I will not do this atm.